src/codegen.cpp

```diff
@@ -2320,6 +2321,9 @@ static jl_cgval_t convert_julia_type(jl_codectx_t &ctx, const jl_cgval_t &v, jl_
         if (v.TIndex && !jl_is_pointerfree(typ)) {
             // discovered that this union-split type must actually be isboxed
             if (v.Vboxed) {
+                if (skip) {
+                    *skip = ctx.builder.CreateNot(emit_exactly_isa(ctx, v, (jl_datatype_t*)typ));
```

Comment:
Isn't Vboxed potentially nullptr at runtime here? Maybe test that tindex==0x80 instead?

Reply:
Yes it is. However, `tindex==0x80` check isn't sufficient here either. Needs an extra case in emit_exactly_isa to check for null (in this case only).